pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -499,6 +501,7 @@ def single_gpu_train(self, model):
     def tpu_train(self, tpu_core_idx, model):
         # put model on tpu
         model.to(xm.xla_device())
+        self.device = xm.xla_device()
```

Comment:
for example here, we could simply override the `.to` in LightningModule and no extra code in the Trainer is necessary

Reply:
the code would be in one place and therefore easier to  maintain

Reply:
also, `.to` calls submodules too, so this approach would automatically take care of nested LightingModules!

Reply:
oh wait... does nn.Module have a .device?? 
if so, i don't think we should overwrite no? i thought the weights had it not the module.

Reply:
i need to sleep lol.. so, i can think about it tomorrow... but sounds interesting :)

Reply:
no it doesn't have it. see here:
https://github.com/pytorch/pytorch/issues/7460
I guess if we do it then we would run into issues when the user starts to move their submodules to different devices by hand. but that would anyway be a problem :) 
yes let's do it tomorrow :)

Reply:
I think we could do this as a read-only property similar to what I did on [metrics](https://github.com/PyTorchLightning/pytorch-lightning/blob/metrics/pytorch_lightning/metrics/metric.py#L61). But I also agree, this should be read-only and not be used for device transfers

Reply:
exactly like you did in metrics, that's exactly what I meant :+1: Nice